packages/utils/src/findSchemaDefinition.ts

```diff
@@ -30,6 +46,31 @@ function findEmbeddedSchemaRecursive<S extends StrictRJSFSchema = RJSFSchema>(sc
   return undefined;
 }
 
+/** Parses a JSONSchema and makes all references absolute with respect to
+ * the `baseURI` argument
+ * @param schema - The schema to be processed
+ * @param baseURI - The base URI to be used for resolving relative references
+ */
+function makeAllReferencesAbsolute<S extends StrictRJSFSchema = RJSFSchema>(schema: S, baseURI: string): S {
+  const currentURI = get(schema, ID_KEY, baseURI);
+  // Make all other references absolute
+  if (REF_KEY in schema) {
+    schema = { ...schema, [REF_KEY]: UriResolver.resolve(currentURI, schema[REF_KEY]!) };
```

Comment:
So we are manipulating the root schema here or the lower-level schema as we are calling `retrieveSchema()`? Honestly, with a few changes to the `createSchemaUtils()` and `Form` in core, we could probably run this code over the `rootSchema` once and cache it in the `schemaUtils` object. And then use that schema within the `Form` by updating the `getRegistry()` function and update the `schema` in the state within `Form` to pick up the cached, updated `rootSchema` from the `schemaUtils`. Does that make sense?

Reply:
Actually, I just updated the `Form` in `core` to always use the `rootSchema` in `schemaUtils` in this [PR](https://github.com/rjsf-team/react-jsonschema-form/pull/4687) so all you have to do is modify the `schemaUtils` to cache the modified `rootSchema` having made references absolute.

Reply:
I moved the logic as requested